File: Server/Scholario.Domain/Entities/GradeWeight.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Scholario.Domain.Entities
{
    public enum GradeWeight
    {
        [Display(Name = "Sprawdzian")]
        Sprawdzian = 5,
        [Display(Name = "Kartkówka")]
        Kartkówka = 4,
        [Display(Name = "Odpowiedź")]
        Odpowiedź = 3,
        [Display(Name = "Praca domowa")]
        PracaDomowa = 2,
        [Display(Name = "Praca dodatkowa")]
        PracaDodatkowa = 1
    }
}

File: Client/Scholario.GUI/src/app/Type/GradeWeight.ts
export const GradeWeight = {
    Sprawdzian: 5,
    Kartkówka: 4,
    Odpowiedź: 3,
    PracaDomowa: 2,
    PracaDodatkowa: 1
  } as const;
  
  // Typy wartości
  export type GradeWeightType = keyof typeof GradeWeight;
  